atom/browser/extensions/atom_extension_system.cc

```diff
@@ -370,6 +377,20 @@ void AtomExtensionSystem::Shared::Observe(int type,
 
       break;
     }
+    case atom::NOTIFICATION_EXTENSION_UNINSTALL_REQUEST: {
+      auto extension =
+        content::Details<const Extension>(details).ptr();
+      base::FilePath install_directory;
+      PathService::Get(component_updater::DIR_COMPONENT_USER,
+          &install_directory);
+      std::string extension_id = extension->id();
+      auto registry = ExtensionRegistry::Get(browser_context_);
+      registry->RemoveEnabled(extension_id);
```

Comment:
not sure if these methods should be called or not, but `ExtensionRegistry::TriggerOnUninstalled` does. See `ExtensionService::UninstallExtension`